### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Updated version of `modern-node-polyfills` includes https://github.com/Aslemammad/modern-node-polyfills/pull/14, which prevents problems like https://github.com/remix-run/remix/issues/6288#issuecomment-1585287474

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,6 +54,9 @@
 		"typescript": "^5.1.3",
 		"vitest": "^0.32.0"
 	},
+	"peerDependencies": {
+		"esbuild": "^0.14.0 || ^0.15.0 || ^0.16.0 || ^0.17.0 || ^0.18.0"
+	},
```
Comment (MichaelDeBoey) [inline package.json:59]: This was always the case, but never specified in `package.json`

File: package.json [hunk 2]
```diff
@@ -84,5 +87,8 @@
 		"ansi-regex": "^5.0.1",
 		"minimist": "^1.2.8"
 	},
-	"packageManager": "yarn@3.6.0"
+	"packageManager": "yarn@3.6.0",
+	"engines": {
+		"node": ">=14.0.0"
+	}
```
Comment (MichaelDeBoey) [inline package.json:93]: This was always the case, but never specified in `package.json`

**Review by imranbarbhuiya — APPROVED** [review]

**Comment by imranbarbhuiya** [comment]
I was trying to add a new commit to update the snapshot but seems like u disabled `Allow edits by maintainers`. Can u allow me to add the commit or u can add the commit too

**Comment by imranbarbhuiya** [comment]
Thanks ❤️

**Comment by MichaelDeBoey** [comment]
@imranbarbhuiya This should have been a `fix` instead of a `chore` as it's actually fixing some problems downstream 🙈

**Pull request merged**